src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java

```diff
@@ -1281,16 +1326,28 @@ private void handleWindow(
         private void handleReset(
             ResetFW reset)
         {
-            doReset(applicationReplyThrottle, applicationReplyId);
             tlsEngine.closeOutbound();
+
+            // RESET is from network reply, so no need to clean that stream
+            doResetApplicationReplyStream();
         }
+
+        private void doResetApplicationReplyStream()
+        {
+            if (applicationReplyThrottle != null)
+            {
+                doReset(applicationReplyThrottle, applicationReplyId);
+            }
+        }
```

Comment:
Rename `doResetApplicationReplyStream` to `handleNetworkReplyDone`.

Reply:
done